### PR TITLE
Chore: Get vault strategy should return strat address

### LINKED ISSUE #104: bug(vaults): getVaultStrategy is not v1.5 compatible
# Summary

Vault v1 and v1.5 have different strategy ABI. This means that using a typed contract for them is not compatible. 

https://github.com/Badger-Finance/badger-sdk/blob/main/src/vaults/vaults.service.ts#L139-L155

The `getVaultStrategy` implementation returns a v1 strategy for every request. Instead, we should just return the address for the strategy and require the user to instantiate (trivially) the version they are expecting as it is anyway a required input. 

## Tasks

- [ ] Update `getVaultStrategy` to return the strategy address
- [ ] Test coverage for `getVaultStrategy`

### PR DESCRIPTION
**Pull request by SHAKOTN**
Update `getVaultStrategy` to return strategy address instead of relevant strategy contract instance

Fixes #104

### CODE REVIEW

File: src/vaults/vaults.service.ts [hunk 1]
```diff
@@ -132,26 +131,28 @@ export class VaultsService extends Service {
...
+      return await vault.strategy();
     }
     const vault = Vault__factory.connect(address, this.sdk.provider);
     const controller = Controller__factory.connect(
       await vault.controller(),
       this.sdk.provider,
     );
-    const strategyAddress = await controller.strategies(await vault.token());
-    return Strategy__factory.connect(strategyAddress, this.sdk.provider);
+    return await controller.strategies(await vault.token());
```
Comment (axejintao) [inline src/vaults/vaults.service.ts:155]: Can drop the await

File: src/vaults/vaults.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    sdk = new BadgerSDK({
+      network: Network.Ethereum,
+      // TODO: Find out how to replace RPC calls with proper mocks
+      provider: 'https://rpc.ftm.tools/',
+    });
+  });
+
+  describe('getVaultStrategy', () => {
+    it('should return v1.5 strategy', async function () {
+      const service = new VaultsService(sdk);
```
Comment (axejintao) [inline src/vaults/vaults.service.spec.ts:19]: You can have 
‘let service’ and use a ‘beforeEach’ to set it each time

**Review by axejintao — APPROVED** [review]

File: src/vaults/vaults.service.ts (hunk 1 above)
Comment (SHAKOTN) [reply src/vaults/vaults.service.ts]: Fixed

File: src/vaults/vaults.service.spec.ts (hunk 2 above)
Comment (SHAKOTN) [reply src/vaults/vaults.service.spec.ts]: Fixed

**Pull request merged**